interfaces/repo.go

```diff
@@ -706,10 +706,11 @@ func (r *Repository) DisconnectSnap(snapName string) ([]*snap.Info, error) {
 		}
 	}
 
-	result := make([]*snap.Info, 0, len(seen))
+	result := make([]string, 0, len(seen))
 	for info := range seen {
-		result = append(result, info)
+		result = append(result, info.Name())
 	}
+	sort.Strings(result)
```

Comment:
Yes, otherwise the order of Setup() calls is not deterministic and one of the tests catches that.